### PR TITLE
[RFC] feat(context-explorer): add a component for context explorer

### PR DESCRIPTION
**Pull request by raymondfeng**
Turning https://github.com/raymondfeng/loopback4-example-inspect into a LoopBack 4 component.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by bajtos** [comment]
> Please make sure to follow all steps outlined in [Adding a new package](https://github.com/strongloop/loopback-next/blob/7fe694865b337c99beb25afc581c5429fb758746/docs/site/DEVELOPING.md#adding-a-new-package). I don't see the new package registered in our docs, CODEOWNERS, etc.

I opened a new issue to automate this check, see https://github.com/strongloop/loopback-next/issues/4673

**Comment by bajtos** [comment]
> coverage/coveralls — Coverage decreased (-0.07%) to 92.436%

@raymondfeng please check the coverage report, are there any important parts of the new code that are missing test coverage?

**Comment by raymondfeng** [comment]
@bajtos Thank you for the comments. I wanted to see if the proposed package has meaningful value to LoopBack 4 first before adding logistics to include it officially - such as adding `CODEOWNER`.

I'll improve the test coverage and docs.

**Comment by raymondfeng** [comment]
We'll incorporate https://github.com/emonddr/loopback4-extension-context-explorer into this extension or another extension such as `context-explorer-ui`.

File: extensions/context-explorer/public/js/graph-util.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  while (nodesToBeAdded.size !== 0) {
+    // Try to find nodes that are successors of starting nodes
+    const matched = new Set();
+    for (const node of nodesToBeAdded) {
+      visited.add(node);
+      const successors = graph.successors(node);
+      for (const s of successors) {
+        matched.add(s);
+      }
+      /*
```
Comment (emonddr) [inline extensions/context-explorer/public/js/graph-util.js]: Old code to delete.

**Comment by raymondfeng** [comment]
@bajtos I added a few tests to cover the new code.

**Comment by achrinza** [comment]
@raymondfeng It's been a while; shall we merge the PR? Since it's still experimental, we'll have many opportunities to add more features and fine-tune it over time before v1.

I think this could be helpful to many newcomers to visualize their LoopBack 4 application, and publishing it to NPM would make it more accessible to others to give feedback.

**Comment by hacksparrow** [comment]
Agree with @achrinza, I am fine with landing this.

**Pull request merged**